### PR TITLE
PUBDEV-8795: Allow S3A to share the built-in AWS credentials providers

### PR DESCRIPTION
**Pull request by michalkurka**

### CODE REVIEW

File: h2o-persist-s3/src/main/java/water/persist/H2OCredentialProviderFactory.java [hunk 1]
```diff
@@ -0,0 +1,76 @@
...
+import java.util.List;
+
+public class H2OCredentialProviderFactory extends CredentialProviderFactory {
+
+    @Override
+    public CredentialProvider createProvider(URI uri, Configuration configuration) {
+        if ("hex".equals(uri.getScheme()) && getClass().getName().equals(uri.getHost()))
+            return new H2OCredentialProvider(new PersistS3.H2OAWSCredentialsProviderChain());
+        else
+            return null;
```
Comment (mn-mikke) [inline h2o-persist-s3/src/main/java/water/persist/H2OCredentialProviderFactory.java:21]: Shouldn't an exception be thrown when it's gets to this state? (Or an error logged at least?)

Comment (michalkurka) [reply h2o-persist-s3/src/main/java/water/persist/H2OCredentialProviderFactory.java:21]: This was intentional - it leaves room for adding additional "h2o" providers. If no provider for "hex://" can be found - the exception will be thrown from the Hadoop code.

Comment (mn-mikke) [reply h2o-persist-s3/src/main/java/water/persist/H2OCredentialProviderFactory.java:21]: ok, thanks for your explanation!

**Review by mn-mikke — APPROVED** [review]
LGTM

**Pull request merged**